frontend/src/app/components/ck-monitor/ck-monitor.component.ts

```diff
@@ -168,6 +214,46 @@ export class CkMonitorComponent implements OnInit, OnDestroy {
     return true;
   }
 
+  async updateTodoItemDataSource(): Promise<void> {
+    const data: TodoItemDisplay[] = [];
+
+    for (const item of this.todoItems) {
+      const date = new Date(`${item.deadline.date} ${item.deadline.time}`);
+      const formattedDate = date.toLocaleDateString('en-CA');
+      const currentDate = new Date();
+      const name = await this.userService.getOneById(item.userID);
```

Comment:
This looks to be pretty expensive, we'd be making a call for each Todo. I would suggest to be able to pass in an optional parameter through the `this.todoItemService.getByProject(projectID)` request which would return a more "expanded" version of each todo where the username is already loaded. We've done this with a couple things already, here's an example: https://github.com/encorelab/ck-board/blob/develop/backend/src/api/workflows.ts#L223

Reply:
Could we move this to another issue to rush the merge for this run or would you advise against going ahead with the merge (currently we have 17 students each with just a couple - probably 1 or 3 - TODO items)?

Reply:
Based on those numbers, I think its fine to merge now!